crates/task-impls/src/events.rs

```diff
@@ -111,4 +111,6 @@ pub enum HotShotEvent<TYPES: NodeType> {
     ///
     /// Like [`HotShotEvent::DAProposalRecv`].
     VidDisperseRecv(Proposal<TYPES, VidDisperse<TYPES>>, TYPES::SignatureKey),
+    /// Upgrade certificate has been received from the network.
+    UpgradeCertificateRecv(UpgradeCertificate<TYPES>),
```

Comment:
Do we also need `UpgradeCertificateSend`, `UpgradeVoteSend`, and `UpgradeVoteRecv`?

Reply:
Yes, I added these!